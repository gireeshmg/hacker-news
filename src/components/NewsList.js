import React from "react";
import ListItem from "./ListItem";
import { Link } from "react-router-dom";
import Chart from "./Chart";

const tableHeader = () => (
  <thead>
    <tr>
      <th>Comments</th>
      <th>Vote Count</th>
      <th>UpVote</th>
      <th className="left-align-text">News Details</th>
    </tr>
  </thead>
);

const NewsList = (props) => {
  const { posts, hideHandler, upVoteHandler } = props;
  return (
    <div>
      <table className="news-table">
        {tableHeader()}
        <tbody>{posts.hits ? posts.hits.map((post) => <ListItem hideHandler={hideHandler} upVoteHandler={upVoteHandler} key={post.objectID} post={post} />) : null}</tbody>
      </table>
      <div className="pagination">

        {posts.page > 1 && (
          <Link to={`/news/${posts.page - 1}`} className="button">
            Previous
          </Link>
        )}
        {posts.nbPages > posts.page + 1 && (
          <Link to={`/news/${posts.page + 1}`} className="button">
            Next
          </Link>
        )}
      </div>

      {posts.hits && <Chart {...posts} />}
    </div>
  );
};

export default NewsList;
