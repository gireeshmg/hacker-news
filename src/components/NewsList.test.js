import React from 'react';
import { render } from '@testing-library/react';
import NewsList from './NewsList';
import { StaticRouter } from 'react-router';

export const posts = {
  "hits": [
    {
      "created_at": "2020-06-07T13:44:03.000Z",
      "title": "Wrath of the Killdozer",
      "url": "https://www.damninteresting.com/the-wrath-of-the-killdozer/",
      "author": "js2",
      "points": 2,
      "story_text": null,
      "comment_text": null,
      "num_comments": 0,
      "story_id": null,
      "story_title": null,
      "story_url": null,
      "parent_id": null,
      "created_at_i": 1591537443,
      "_tags": [
        "story",
        "author_js2",
        "story_23447271"
      ],
      "objectID": "23447271",
      "_highlightResult": {
        "title": {
          "value": "Wrath of the Killdozer",
          "matchLevel": "none",
          "matchedWords": []
        },
        "url": {
          "value": "https://www.damninteresting.com/the-wrath-of-the-killdozer/",
          "matchLevel": "none",
          "matchedWords": []
        },
        "author": {
          "value": "js2",
          "matchLevel": "none",
          "matchedWords": []
        }
      }
    },
    {
      "created_at": "2020-06-07T13:41:25.000Z",
      "title": "There's a growing call to defund the police. Here's what it means",
      "url": "https://www.cnn.com/2020/06/06/us/what-is-defund-police-trnd/index.html",
      "author": "MilnerRoute",
      "points": 16,
      "story_text": null,
      "comment_text": null,
      "num_comments": 0,
      "story_id": null,
      "story_title": null,
      "story_url": null,
      "parent_id": null,
      "created_at_i": 1591537285,
      "_tags": [
        "story",
        "author_MilnerRoute",
        "story_23447259",
        "front_page"
      ],
      "objectID": "23447259",
      "_highlightResult": {
        "title": {
          "value": "There's a growing call to defund the police. Here's what it means",
          "matchLevel": "none",
          "matchedWords": []
        },
        "url": {
          "value": "https://www.cnn.com/2020/06/06/us/what-is-defund-police-trnd/index.html",
          "matchLevel": "none",
          "matchedWords": []
        },
        "author": {
          "value": "MilnerRoute",
          "matchLevel": "none",
          "matchedWords": []
        }
      }
    }
  ],
  "nbHits": 2093621,
  "page": 1,
  "nbPages": 50,
  "hitsPerPage": 20,
  "exhaustiveNbHits": false,
  "query": "",
  "params": "advancedSyntax=true&analytics=true&analyticsTags=backend&page=1&tags=story",
  "processingTimeMS": 1
};

const hideHandler = () => {
  console.log('hideHandler clicked');
}

const upVoteHandler = () => {
  console.log('upVoteHandler clicked');
}

test('renders List items', () => {
  const { container } = render(<StaticRouter>
    <NewsList hideHandler={hideHandler} upVoteHandler={upVoteHandler} posts={posts} />
  </StaticRouter>
  );
  expect(container.firstChild).toMatchSnapshot();
});
