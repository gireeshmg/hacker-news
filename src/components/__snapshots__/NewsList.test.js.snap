// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List items 1`] = `
<div>
  <table
    class="news-table"
  >
    <thead>
      <tr>
        <th>
          Comments
        </th>
        <th>
          Vote Count
        </th>
        <th>
          UpVote
        </th>
        <th
          class="left-align-text"
        >
          News Details
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="center-align-text"
        >
          0
        </td>
        <td
          class="center-align-text"
        >
          2
        </td>
        <td
          class="center-align-text"
        >
          <button
            class="btn-upvote"
          >
            Up vote
          </button>
        </td>
        <td>
          <a
            class="news-anchor"
            href="https://www.damninteresting.com/the-wrath-of-the-killdozer/"
          >
            Wrath of the Killdozer
          </a>
          <span
            class="domain-name"
          >
            (damninteresting.com)
             by
          </span>
           
          <span
            class="author"
          >
            js2
          </span>
          <span
            class="time"
          >
            2 hours ago
          </span>
          <button
            class="btn-hide"
          >
            [ hide ]
          </button>
        </td>
      </tr>
      <tr>
        <td
          class="center-align-text"
        >
          0
        </td>
        <td
          class="center-align-text"
        >
          16
        </td>
        <td
          class="center-align-text"
        >
          <button
            class="btn-upvote"
          >
            Up vote
          </button>
        </td>
        <td>
          <a
            class="news-anchor"
            href="https://www.cnn.com/2020/06/06/us/what-is-defund-police-trnd/index.html"
          >
            There's a growing call to defund the police. Here's what it means
          </a>
          <span
            class="domain-name"
          >
            (cnn.com)
             by
          </span>
           
          <span
            class="author"
          >
            MilnerRoute
          </span>
          <span
            class="time"
          >
            2 hours ago
          </span>
          <button
            class="btn-hide"
          >
            [ hide ]
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="pagination"
  >
    <a
      class="button"
      href="/news/2"
    >
      Next
    </a>
  </div>
  <div
    class="chart-container"
  >
    <div
      id="chart-1"
      style="height: 300px; line-height: 300px; width: 100%; text-align: center; color: rgb(153, 153, 153); font-size: 14px; font-family: 'Lucida Grande', 'Lucida Sans Unicode', Verdana, Arial, Helvetica, sans-serif;"
    >
      <canvas />
    </div>
  </div>
</div>
`;
